lib/server.js

```diff
@@ -718,7 +718,7 @@ WalletService.prototype._getUtxosForAddresses = function(addresses, cb) {
       var u = _.pick(utxo, ['txid', 'vout', 'address', 'scriptPubKey', 'amount', 'satoshis', 'confirmations']);
       u.confirmations = u.confirmations || 0;
       u.locked = false;
-      u.satoshis = u.satoshis ? +u.satoshis : Utils.strip(u.amount * 1e8);
+      u.satoshis = _.isNumber(u.satoshis) ? +u.satoshis : Utils.strip(u.amount * 1e8);
```

Comment:
Good catch! As an additional measure I added a PR to ignore UTXOs with amount 0: https://github.com/troggy/bitcore-wallet-service/pull/2